### ISSUE TITLE
#12483 does not solve the problem for chrome on iOS

### ISSUE THREAD

**Issue by leon** [BabylonJS/Babylon.js#12507]
https://github.com/BabylonJS/Babylon.js/pull/12483

Both Chrome and Safari seem to be suffering from the problem with webgl2 and antialiasing.
And It's probably not limited to these two. since as I understand it iOS and iPadOS do not support other browsers, instead the browsers are simply wrappers around the WebKitView Engine.

Therefor we need to take into account many more userAgent strings.

Here are the different user agent strings for iOS and iPadOS for both chrome and safari for reference.

```

// iOS 15.4.1 Chrome 101
Mozilla/5.0 (iPhone; CPU iPhone OS 15_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/101.0.4951.44 Mobile/15E148 Safari/604.1
// iOS 15.4.1 Safari 15.4.1
Mozilla/5.0 (iPhone; CPU iPhone OS 15_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Mobile/15E148 Safari/604.1

// iPadOS 15.4.1 Chrome 101
Mozilla/5.0 (iPad; CPU OS 15_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/101.0.4951.44 Mobile/15E148 Safari/604.1
// iPadOS 15.4.1 Safari 15.4.1
Mozilla/5.0 (iPad; CPU OS 15_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Mobile/15H321 Safari/604.1
```

The problem lies with that the userAgent string for chrome looks different and does not match the regex we currently have.

# Solution
I would suggest we do the match against the first part of the string instead since that looks the same in both and do an extra check that the engine is AppleWebKit.

also we don't need the `/g` flag since we are only looking for the first occurrence of the regex.

```
const fixYourBugsAlreadyRegexp = /OS\s(\d\d)_(\d).+AppleWebKit/;
```

**Comment by RaananW** [comment]
Yeah, I totally forgot how the apple ecosystem works. Chrome uses Safari to render.

I am not entirely happy about changing it to all UAs using AppleWebKit. AFAIK (and please correct me if I am wrong), this is an issue with 15.4.X version of safari. earlier versions work well. Is that right?

**Comment by leon** [comment]
I only meant changing the regex. 

The new regex contains the same groups so the check for version 15 and 4 still apply.

**Comment by RaananW** [comment]
got it! just saw the version signature, sorry.  I'll run a few tests on desktop OSX as well just to be sure. thanks

**Comment by leon** [comment]
To my knowledge it doesn't affect desktop, only mobile safari.

So there is no need for the regex to match desktop

**Comment by RaananW** [comment]
I don't have any apple devices, I am only checking with ua-overrides. AFAIK there was an issue with desktops as well.  I'll ask internally to make sure :-)
